pkg/gateway/map.go

```diff
@@ -0,0 +1,27 @@
+package gateway
+
+import (
+	"sync/atomic"
+)
+
+type Map struct {
+	clustersMap atomic.Value
+}
+
```

Comment:
I don’t think the `Map` struct really adds any value - it just wraps `Atomic.Value` so might as well use that directly.  Another idea is to make it an interface in the plugin package, ie:

```
type ClusterStatus interface {
    IsConnected(string clusterID), bool
}
```

The GW controller would implement it. This has the advantage of de-coupling the plugin from the GW in case we implement it in a different way in the future.